lib/localparticipant.js

```diff
@@ -128,7 +128,9 @@ class LocalParticipant extends Participant {
    * @returns {void}
    */
   _addLocalTrack(track, priority) {
-    this._signaling.addTrack(track._trackSender, track.name, priority);
+    // check if track has noise cancellation enabled.
+    const vendor = track?.noiseCancellation?.vendor;
```

Comment:
Optional chaining is needed since `track.noiseCancellation` can be null. But I think your question is valid - we don't  need it for track I will change to 

`track?.noiseCancellation?.vendor` => `track.noiseCancellation?.vendor`